lib/iris/_deprecation.py

```diff
@@ -32,8 +32,34 @@ class IrisDeprecation(UserWarning):
     pass
 
 
-def warn_deprecated(msg, **kwargs):
-    warnings.warn(msg, IrisDeprecation, **kwargs)
+def warn_deprecated(msg, stacklevel=2):
+    """
+    Issue an Iris deprecation warning.
+
+    Calls :func:`warnings.warn', to emit the message 'msg' as a
+    :class:`warnings.warning`, of the subclass :class:`IrisDeprecationWarning`.
+
+    The 'stacklevel' keyword is passed through to warnings.warn.  However by
+    default this is set to 2, which ensures that the identified code line is in
+    the caller, rather than in this routine.
+    See :mod:`warnings` module documentation.
+
+    For example::
+
+        >>> from iris._deprecation import warn_deprecated
+        >>> def arrgh():
+        ...    warn_deprecated('"arrgh" is deprecated since version 3.5.')
+        ...    return 1
+        ...
+        >>> print arrgh()
```

Comment:
I thought we were trying to maintain Python 3 compatibility everywhere, including doctests/documentation. `print(arrgh())` vs `print arrgh()`, doesn't look that bad to me, the function name probably is more to blame than the parentheses!